src/capymoa/regressor/_soknl.py

```diff
@@ -1,17 +1,40 @@
 # Library imports
-from typing import Optional, Union
+
+from ._soknl_base_tree import SOKNLBT
+from moa.classifiers.meta import SelfOptimisingKNearestLeaves as _MOA_SOKNL
 
 from capymoa.base import (
     MOARegressor, _extract_moa_learner_CLI
 )
 
-from capymoa.splitcriteria import SplitCriterion, _split_criterion_to_cli_str
-from capymoa.stream._stream import Schema
-from moa.classifiers.meta import SelfOptimisingKNearestLeaves as _MOA_SOKNL
-from ._soknl_base_tree import SOKNLBT
-
 
 class SOKNL(MOARegressor):
+    """Self-Optimising K-Nearest Leaves (SOKNL) Implementation.
+
+    SOKNL is an extension to AdaptiveRandomForestRegressor.
+    SOKNL stores abstract information for all instance seen by the leaves, use them to calculate the distance
+    from the leaf to a certain instance. Then final predictions are yielded by the closest k leaves in the forest.
+
```

Comment:
Before the reference add a "See also" for ARFReg and MOARegressor. You can refer to other classes using the syntax below (See _adaptive_random_forest.py for reference on the syntax as well)

See also :py:class:\`capymoa.regressor.AdaptiveRandomForestRegressor\`
See :py:class:\`capymoa.base.MOARegressor\` for train and predict.